src/include/stir/recon_buildblock/ML_estimate_component_based_normalisation.h

```diff
@@ -14,25 +15,171 @@
  */
 #include "stir/common.h"
 #include <string>
+#include "stir/ProjData.h"
+#include "stir/ML_norm.h"
 
 START_NAMESPACE_STIR
 
-class ProjData;
-
 /*!
- \brief Find normalisation factors using a maximum likelihood approach
-
+  \brief Find normalisation factors using a maximum likelihood approach
   \ingroup recon_buildblock
+        \param out_filename_prefix The prefix for the output files
+        \param measured_data The measured projection data
+        \param model_data The model projection data
+        \param num_eff_iterations The number of (sub-)efficiency iterations to perform per iteration of the algorithm
+        \param num_iterations The number of algorithm iterations to perform
+        \param do_geo Whether to perform geo normalization calculations
+        \param do_block Whether to perform block normalization calculations
+        \param do_symmetry_per_block Whether to perform block normalization calculations with symmetry
+        \param do_KL Whether to perform Kullback-Leibler divergence calculations and display the KL value. This can be slow.
+        \param do_display Whether to display the progress of the algorithm.
+        \param do_save_to_file Whether to save the each iteration of the efficiencies, geo data and block data to file.
 */
 void ML_estimate_component_based_normalisation(const std::string& out_filename_prefix,
-                                               const ProjData& measured_data,
-                                               const ProjData& model_data,
+                                               const std::shared_ptr<ProjData>& measured_data,
+                                               const std::shared_ptr<ProjData>& model_data,
```

Comment:
if we use `shared_ptr`, please rename to `measured_data_sptr`. A question though is if we need to (i.e. does the object need access to these after the constructor?)

Reply:
Correct, the member is not used outside of the constructor. I have removed it.